### PR TITLE
Use fluxible-addons-react temporarily for backwards compatibility; Wa…

### PR DESCRIPTION
**Pull request by mridgway**
…rn on usage of addons directly from fluxible

This removes Fluxible's dependency on React. Fluxible will still have a dependency on fluxible-addons-react for one minor version with warnings if the addons are used directly from `fluxible/addons`. This will allow us to completely remove the React dependency from the dep chain in the next version.

The plan at this point is to 1.0 Fluxible after this upcoming minor version.

### CODE REVIEW

**Comment by redonkulus** [comment]
:+1:  after fluxible.io is updated.

**Comment by mridgway** [comment]
Just found some things to fix:

 - [ ] npm 1.x fails on peer dependency
 - [ ] Add warnings for `fluxible/addons` usage.

File: addons/provideContext.js [hunk 1]
```diff
@@ -1,80 +1,6 @@
...
-
-        render: function () {
-            return React.createElement(Component, this.props);
-        }
-    });
-
-    hoistNonReactStatics(ContextProvider, Component);
-
-    return ContextProvider;
+if (process.env.NODE_ENV === 'production') {
```
Comment (Vijar) [inline addons/provideContext.js]: Shouldn't we warn in dev?

Comment (mridgway) [inline addons/provideContext.js]: Yes. Fixing.

**Comment by Vijar** [comment]
Build will fail because `fluxible-addons-react` depends on this PR being published...so lets :shipit: .

**Pull request merged**